annotation/consume_trigger.go

```diff
@@ -134,11 +203,19 @@ func (PtrLoadPrestring) String() string {
 //
 // note: this trigger is produced only if config.ErrorOnNilableMapRead == true
 type MapAccess struct {
-	ConsumeTriggerTautology
+	*ConsumeTriggerTautology
+}
+
+// equals returns true if the passed ConsumingAnnotationTrigger is equal to this one
+func (i *MapAccess) equals(other ConsumingAnnotationTrigger) bool {
+	if other, ok := other.(*PtrLoad); ok {
```

Comment:
Good catch! I agree that we should add unit tests for catching such inadvertent mistakes. Added unit tests for checking implementations of `equals` method of `ConsumingAnnotationTrigger`, `ProducingAnnotationTrigger`, and `Key` interface. The unit tests check for (1) and (2), since (3) would prove to be quite tedious to write the test cases.

Also, as discussed offline, reflection could not be used for this purpose proved. Hence, for ensuring that we have unit tests for all the structs implementing the given interface, I resorted to the following:
- expected: extract all structs implementing the interface using `types` package
- actual: extract all structs initialized in the test case
- compare `expected` and `actual` to find and report missing structs, if any